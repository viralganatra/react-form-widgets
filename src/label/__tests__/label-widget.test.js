import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import LabelWidget from '../label-widget';

describe('LabelWidget', () => {
    it('should render a single label widget correctly', () => {
        const wrapper= shallow(<LabelWidget>Test</LabelWidget>);

        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should throw an error if there are no children', () => {
        const error = console.error;
        console.error = jest.fn();

        shallow(<LabelWidget />);

        expect(console.error).toBeCalled();

        console.error = error;
    });

    it('should apply any label props to the label element', () => {
        const wrapper = shallow(
            <LabelWidget className="test" htmlFor="test">Test</LabelWidget>,
        );

        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
});
