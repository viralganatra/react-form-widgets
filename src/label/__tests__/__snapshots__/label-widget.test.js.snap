exports[`LabelWidget should apply any label props to the label element 1`] = `
<label
  className="test"
  htmlFor="test">
  Test
</label>
`;

exports[`LabelWidget should render a single label widget correctly 1`] = `
<label>
  Test
</label>
`;
